dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/WebContainer.java

```diff
@@ -1568,41 +1580,56 @@ protected void unsetCacheServletWrapperFactory(CacheServletWrapperFactory factor
-    protected synchronized void setVersion(ServiceReference<ServletVersion> reference) {
+   /*
+    * Should only be called by loadServletSpecVersion
+    */
+    protected static void setVersion(int version){
         String methodName = "setVersion";
-        versionRef = reference;
-        WebContainer.loadedContainerSpecLevel = (Integer) reference.getProperty("version");
+        WebContainer.loadedContainerSpecLevel = version;
         if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
             Tr.debug(tc, methodName, "loadedContainerSpecLevel [ " + WebContainer.loadedContainerSpecLevel + " ]");
         }
     }
 
-    protected synchronized void unsetVersion(ServiceReference<ServletVersion> reference) {
-        if (reference == this.versionRef) {
-            versionRef = null;
-            WebContainer.loadedContainerSpecLevel = DEFAULT_MAX_VERSION;
-        }
+   /*
+    * Not sure if this should exist or if it should be called when webcontainer is deactivating? 
+    */
+    protected static void unsetVersion(int version){
+        WebContainer.loadedContainerSpecLevel = DEFAULT_MAX_VERSION;
     }
-    
-    public static final int SPEC_LEVEL_UNLOADED = -1;
-    public static final int SPEC_LEVEL_30 = 30;
-    public static final int SPEC_LEVEL_31 = 31;
-    public static final int SPEC_LEVEL_40 = 40;
-    public static final int SPEC_LEVEL_50 = 50;
-    private static final int DEFAULT_SPEC_LEVEL = 30;
-    
-    private static int loadedContainerSpecLevel = SPEC_LEVEL_UNLOADED;
-    
+
+    protected static synchronized void loadServletVersion(){
+        String methodName = "loadServletVersion";
+
+        try (InputStream input = WebContainer.class.getClassLoader().getResourceAsStream("com/ibm/ws/webcontainer/speclevel/servletSpecLevel.properties")) {
+
+            // null check fixes errors in wc unit tests 
+            if(input != null){
+                Properties prop = new Properties();
+                prop.load(input);
+                setVersion(Integer.parseInt(prop.getProperty("version")));
+            } else {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, methodName, "InputStream was null for servletSpecLevel.properties");
+                }
+            }
+
+        } catch (Exception ex) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, methodName, "Exception occured: " + ex.getCause());
+            }
+        } 
+
+    }
+        
     public static int getServletContainerSpecLevel() {
         String methodName = "getServletContainerSpecLevel";
 
         if (WebContainer.loadedContainerSpecLevel == SPEC_LEVEL_UNLOADED) {
             CountDownLatch currentLatch = selfInit;
             // wait for activation
             try {
-                currentLatch.await(5, TimeUnit.SECONDS);
+                currentLatch.await(2, TimeUnit.SECONDS);
```

Comment:
Why are we decreasing the `await` time here?

Reply:
Is it still necessary as we're not waiting on the component to activate now are we? Lets discuss.

Reply:
@tjwatson, you set it to 5 seconds here originally: https://github.com/OpenLiberty/open-liberty/commit/8623a34c27159842255846f4f5e5613c12bbe7bd#diff-4139058e3df37a2fdb3b608f7eecbc69b3b5b4b8537d61717e80b88acf3368beR611-R623

Would you have any opinion?

Reply:
I do not see any point of this logic any more.  Just have the method simply be:

```java
    return WebContainer.loadedContainerSpecLevel;
```